File: app/src/main/java/com.example.safehome/Sensors/Sensor.java
package com.example.safehome.Sensors;

public interface Sensor <Object> {

    /**
     * detects whether an event has occurred
      */
    boolean isEventDectected();

    /**
     * identifies the type of sensor
     */
    void identify();


    /**
     * enables notifications
     */
    void enableNotification();

    /**
     * disables notifications
     */
    void disableNotification();

    /**
     * reconfigures sensor
     */
    void reconfigure();

    /**
     * adds sensor to system
     */
    void addSensorMomento();

    /**
     * gets existing sensor from system
     */
    void getSensorMomento();
}

File: app/src/main/kotlin/com.example.safehome/Sensors/Sensor.kt
package com.example.safehome.Sensors

interface Sensor{


    /**
     * detects whether an event has occurred
     */
    fun isEventDetected(): Boolean

    /**
     * identifies the type of sensor
     */
    fun identify()


    /**
     * enables notifications
     */
    fun enableNotification()

    /**
     * disables notifications
     */
    fun disableNotification()

    /**
     * reconfigures sensor
     */
    fun reconfigure()

    /**
     * adds sensor to system
     */
    fun addSensorMomento()

    /**
     * gets existing sensor from system
     */
    fun getSensorMomento()

}